src/main/java/morgana/exceptions/MorganaException.java

```diff
@@ -0,0 +1,17 @@
+package morgana.exceptions;
+
+/**
+ * Represents a custom exception used in the Morgana application.
+ * This exception is thrown when specific errors related to the application's
+ * functionality occur.
+ */
+public class MorganaException extends Exception {
+    /**
+     * Constructs a {@code MorganaException} with the specified detail message.
+     *
+     * @param message The detail message describing the specific error that occurred.
+     */
+    public MorganaException(String message) {
```

Comment:
Would the code be more readable if different error types are extracted?

Reply:
You're right. Thank you for the suggestion.